### PR TITLE
feat: disabled 2fa feature

### PR DESCRIPTION
**Pull request by gorjan5sk**
- use new snjs mfa method to check if 2fa feature is available

### CODE REVIEW

File: app/assets/javascripts/preferences/PreferencesViewWrapper.tsx [hunk 1]
```diff
@@ -18,7 +18,8 @@ export const PreferencesViewWrapper: FunctionComponent<PreferencesViewWrapperPro
       <PreferencesView
         closePreferences={() => appState.preferences.closePreferences()}
         application={application}
-        mfaGateway={application}
+        mfaProvider={application}
+        userProvider={application}
```
Comment (vardan-arm) [inline app/assets/javascripts/preferences/PreferencesViewWrapper.tsx:22]: We use 2 separate types for `mfaProvider` and `userProvider` but pass the same `application` object to them. Probably it's better to either use one prop for `application` with appropriate name or pass application's respective fields to `mfaProvider` and `userProvider` (e.g., pass only `application.getUser` to `userProvider`)

File: app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorAuth.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export const is2FADisabled = (s: TwoFactorStatus): s is 'two-factor-disabled' =>
   s === 'two-factor-disabled';
 
 export const is2FAActivation = (s: TwoFactorStatus): s is TwoFactorActivation =>
```
Comment (vardan-arm) [inline app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorAuth.ts:13]: Let's use more meaningful name for `s` :)

File: app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorAuthView.tsx [hunk 3]
```diff
@@ -7,45 +7,38 @@ import {
...
+        <div className="flex flex-row items-center">
+          <div className="flex-grow flex flex-col">
+            <Title>Two-factor authentication</Title>
+            <Text>
+              An extra layer of security when logging in to your account.
+            </Text>
+          </div>
+          <Switch
+            checked={!is2FADisabled(auth.status)}
+            onChange={() => auth.toggle2FA()}
```
Comment (vardan-arm) [inline app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorAuthView.tsx:28]:
```suggestion
            onChange={auth.toggle2FA}
```

File: app/assets/javascripts/preferences/PreferencesViewWrapper.tsx (hunk 1 above)
Comment (gorjan5sk) [reply app/assets/javascripts/preferences/PreferencesViewWrapper.tsx:22]: Separation of concerns. We can test  components that only depend on x:getUser(), that only depend on MfaProvider stuff, without mocking the whole WebApplication along with the kitchen sink :D

File: app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorAuth.ts (hunk 2 above)
Comment (gorjan5sk) [reply app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorAuth.ts]: Done.

File: app/assets/javascripts/preferences/PreferencesViewWrapper.tsx (hunk 1 above)
Comment (vardan-arm) [reply app/assets/javascripts/preferences/PreferencesViewWrapper.tsx:22]: It looks very strange tbh :D But I think we'll have better (practical) understanding of pros and cons of this approach only after writing some tests :)

Comment (gorjan5sk) [reply app/assets/javascripts/preferences/PreferencesViewWrapper.tsx:22]: They're coming! :D 
Can I get an approval now? :)

**Review by vardan-arm — APPROVED** [review]

**Pull request merged**